exporter/exporterhelper/internal/obs_report_sender.go

```diff
@@ -137,3 +162,38 @@ func toNumItems(numExportedItems int, err error) (int64, int64) {
 	}
 	return int64(numExportedItems), 0
 }
+
+func extractFailureAttributes(err error) attribute.Set {
+	if err == nil {
+		return attribute.NewSet()
+	}
+
+	attrs := []attribute.KeyValue{}
+	reason := determineFailureReason(err)
+	attrs = append(attrs, attribute.String(string(FailureReasonKey), reason))
+	isPermanent := consumererror.IsPermanent(err)
+	attrs = append(attrs, attribute.Bool(string(FailurePermanentKey), isPermanent))
+	return attribute.NewSet(attrs...)
+}
+
+func determineFailureReason(err error) string {
```

Comment:
I wonder if we should support gRPC status more generally? Aside from "shutdown" and the ones I suggest removing, the proposed reasons below are a subset of gRPC status codes.

Also, with that in mind, should we be using the [`error.type`](https://opentelemetry.io/docs/specs/semconv/registry/attributes/error/#error-type) SemConv attribute? i.e. the type of the error that led to the failure.